scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,288 @@
+-----------------------------------
+--  PET: Elementals
+-----------------------------------
+require("scripts/globals/monstertpmoves")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+require("scripts/globals/utils")
+require("scripts/globals/weather")
+
+function onMobDeath(mob, player, isKiller)
+
+end;
+
+-- on engage, return the delay reduction based on skill to cast on battle start
+function onPetEngage(pet,delay)
+    -- return incoming delay value to return to legacy/monster mode    
+    local master = pet:getMaster()    
+    local fastCast = 0
+    if (pet:getSystem() == 11) then      -- elemental spirit   
+        local smnSkill = getSummoningSkillOverCap(pet)
```

Comment:
Is smnSkillOverCap used for anything other than the fastCast chance at the start? (Not necessarily right now, but do spirits on retail use it for other things?) If not, could combine the two functions into a single function.
Here:
```
    fastCast, overCap = getTimeCost(pet)
```

Then in getTimeCost():
```
    return timeCost, math.max(summoningSkill - maxSkill, 0)
```